openmrs/src/main/java/org/motechproject/openmrs/resource/impl/EncounterResourceImpl.java

```diff
@@ -27,32 +30,33 @@ public EncounterResourceImpl(RestOperations restOperations) {
 
     @Override
     public Encounter createEncounter(Config config, Encounter encounter) {
-        String requestJson = buildGsonWithAdapters().toJson(encounter);
+        String requestJson = buildGsonWithAdaptersSerialize().toJson(encounter);
         String responseJson = postForJson(config, requestJson, "/encounter?v=full");
 
-        Encounter createdEncounter = (Encounter) JsonUtils.readJson(responseJson, Encounter.class);
-        setProviderFromEncounterProviderList(createdEncounter);
-        return createdEncounter;
+        return checkVersionAndSetEncounter(config, responseJson);
     }
 
     @Override
     public EncounterListResult queryForAllEncountersByPatientId(Config config, String id) {
         String responseJson = getJson(config, "/encounter?patient={id}&v=full", id);
-        
-        EncounterListResult encounterList = (EncounterListResult) JsonUtils.readJson(responseJson, EncounterListResult.class);
-        for(Encounter encounter : encounterList.getResults()) {
-            setProviderFromEncounterProviderList(encounter);
+
+        List<Encounter> encounterList = new ArrayList<>();
+        JsonArray allEncounters = prepareJsonArray(responseJson);
```

Comment:
rename prepareJsonArray function to prepareAllEncountersList and then remove additional allEncounters variable and call that function inside for declaration